### DISCUSSION TITLE
Getting all pom files used during `updateFull`

### DISCUSSION THREAD

**Discussion by timbertson** [sbt/sbt#7451] [Q&A]
I'm working on a plugin which supports building sbt projects in an offline, reproducible way using [nix](https://nixos.org/). It's a bit like https://github.com/zaninime/sbt-derivation, but with a different underlying approach.

The big thing I want to do is to determine the minimal set of files I need to populate a minimal coursier cache, and then use that cache to build fully offline. I've got a lot of it working based on `updateFull.value`, then iterating over all the configurations / modules / artifacts in the resulting `UpdateReport` to extract all the used artifacts.

The one thing I'm struggling with is figuring out what _pom_ files are needed in the cache for the resolution process itself (I haven't looked into ivy files yet). For modules with a `.jar` I'm taking a hacky approach where I see if there's an adjacent `.pom` file in the cache and include it. But this breaks down on meta-modules like [error_prone_parent](https://mvnrepository.com/artifact/com.google.errorprone/error_prone_parent/2.23.0) which don't have any artifacts themselves. I end up missing their `.pom` from the cache, which causes resolution to fail.

It doesn't look like `updateFull` includes any details on these meta-modules at all. Is there any way I can configure it to include them (or better: report all `.pom`s as first-class artifacts)? Or are the details of pom files hidden somewhere in coursier and not exposed by SBT?

**Comment by eed3si9n** [comment]
> Or are the details of pom files hidden somewhere in coursier and not exposed by SBT?

My guess is that POM (including parent POM) files are details of Coursier.
